geomesa-core/src/main/scala/org/locationtech/geomesa/core/index/FilterHelper.scala

```diff
@@ -73,6 +73,22 @@ object FilterHelper {
       ff.or(filterList)
   }
 
+  def isFilterWholeWorld(f: Filter): Boolean = f match {
+      case op: BBOX       => isOperationGeomWholeWorld(op)
+      case op: Within     => isOperationGeomWholeWorld(op)
+      case op: Intersects => isOperationGeomWholeWorld(op)
+      case op: Overlaps   => isOperationGeomWholeWorld(op)
+      case _ => false
+    }
+
+  private def isOperationGeomWholeWorld[Op <: BinarySpatialOperator](op: Op): Boolean = {
+    val e2 = op.getExpression2.asInstanceOf[Literal]
```

Comment:
i don't think expresssion 2 is guaranteed to be the literal... might want to use the checkOrder function